platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,163 @@
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
+            logger.log_info(f"{self.get_name()}: Going Down Unsuccessful")
+            self.dpu_power_off(forced=True)
+            return None
+        return True
+
+    def dpu_power_off(self, forced=False):
+        """Per DPU Power off API"""
+        logger.log_info(f"{self.get_name()}: Power off forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "1")
+            logger.log_info(f"{self.get_name()}: Force Power Off complete")
+            return True
+        self.write_file(self.set_go_down_path, "1")
+        if not self.dpu_go_down(): 
+            """If go_down returns None, 
+            that means Force power off is executed, No need to power off"""
+            return True
+        self.write_file(self.set_pwr_path, "1")
+        logger.log_info(f"{self.get_name()}: Power Off complete")
+        return True
+
+    def _power_on_force(self, count=4):
+        """Per DPU Power on with force private function"""
+        if count < 4:
+            logger.log_info(f"{self.get_name()}: Failed Force Power on! Retry {4-count}..")
+        self.write_file(self.set_pwr_f_path, "0")
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        if not dpu_rdy:
+            if count > 1:
+                time.sleep(1)
+                return self._power_on_force(count=count - 1)
+            logger.log_info(f"{self.get_name()}: Failed Force power on! Exiting")
+            return False
+        logger.log_info(f"{self.get_name()}: Force Power on Successful!")
+        return True
+
+    def _power_on(self):
+        """Per DPU Power on without force private function"""
+        self.write_file(self.set_pwr_path, "0")
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        logger.log_info(f"Return value is {dpu_rdy}")
+        if not dpu_rdy:
+            logger.log_info(f"{self.get_name()}: Failed power on! Trying Force Power on")
+            return self._power_on_force()
+        logger.log_info(f"{self.get_name()}:Power on Successful!")
+        return True
+
+    def dpu_power_on(self, forced=False):
+        """Per DPU Power on API"""
+        logger.log_info(f"{self.get_name()}: Power on with force = {forced}")
+        if forced:
+            return self._power_on_force()
+        else:
+            return self._power_on()
+
+    def dpu_reboot(self):
+        """Per DPU Reboot API"""
+        logger.log_info(f"{self.get_name()}: Reboot")
+        self.write_file(self.set_go_down_path, "1")
+        self.dpu_go_down()
```

Comment:
Can we put  self.write_file(self.set_go_down_path, "1") and self.write_file(self.set_go_down_path, "0") inside the self.dpu_go_down()?

Reply:
`self.write_file(self.set_go_down_path, "1")`  is added back to `dpu_go_down` we are not required to set ` self.write_file(self.set_go_down_path, "0")` each time we call `dpu_go_down` (we do not set it in `dpu_power_off` so not adding that to the function)